### PR TITLE
Add with_unit method for metrics API

### PR DESCRIPTION
**Pull request by ozerovandrei**
Add `with_unit` method to `CounterBuilder`, `SumObserverBuilder`,
`UpDownSumObserverBuilder`, `ValueObserverBuilder` structs.

Add `set_unit` method to `Descriptor` struct.

For #276

### CODE REVIEW

**Review by TommyCpp — APPROVED** [review]
LGTM 👍 Thanks!

File: opentelemetry/src/metrics/descriptor.rs [hunk 1]
```diff
@@ -73,6 +74,11 @@ impl Descriptor {
         self.config.unit.as_ref().map(|unit| unit.as_ref())
     }
 
+    /// Assign a new unit.
+    pub fn set_unit(&mut self, unit: Unit) {
```
Comment (jtescher) [inline opentelemetry/src/metrics/descriptor.rs:78]: Hm we probably don't want this in the public api. Maybe marking the descriptor properties as `pub(crate)` would be simpler?

Comment (ozerovandrei) [reply opentelemetry/src/metrics/descriptor.rs]: I removed this method and made the `config` property `pub(crate)` to access it directly in builders.

**Review by jtescher — APPROVED** [review]
Looks good thanks @ozerovandrei

**Pull request merged**